include/pmacc/assert.hpp

```diff
@@ -55,4 +59,16 @@
  */
 #    define PMACC_ASSERT_MSG(expr, msg) (!!(expr)) ? ((void) 0) : pmacc::abortWithError(#    expr, __FILE__, __LINE__, msg)
 
+/** assert check
```

Comment:
Maybe it's useful to extend a comment saying this one is for in-kernel use, and for the previously existing assert add that it's for out-of-kernel use? Otherwise it maybe unclear if someone just glanceds through this file